solutions/swb-reference/integration-tests/tests/isolated/environmentTypeConfigs/list.test.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+import ClientSession from '../../../support/clientSession';
+import Setup from '../../../support/setup';
+import HttpError from '../../../support/utils/HttpError';
+import { checkHttpError } from '../../../support/utils/utilities';
+
+describe('list environment type configs', () => {
+  const setup: Setup = new Setup();
+  let adminSession: ClientSession;
+  const envTypeId = setup.getSettings().get('envTypeId');
+
+  beforeEach(() => {
+    expect.hasAssertions();
+  });
+
+  beforeAll(async () => {
+    adminSession = await setup.getDefaultAdminSession();
+  });
+
+  afterAll(async () => {
+    await setup.cleanup();
+  });
+
+  test('list environments type configs when filter and sorting by name', async () => {
+    const { data: response } = await adminSession.resources.environmentTypeConfigs.get({}, envTypeId);
+    expect(Array.isArray(response.data)).toBe(true);
+  });
+
+  test('list environments type configs fails when using invalid format envType Id', async () => {
+    try {
+      await adminSession.resources.environmentTypeConfigs.get({}, 'invalid-envType-id');
```

Comment:
Hmm, interesting pattern. How I've seen it done is you can nest ETC inside of ET. An example is shown [here](https://github.com/awslabs/service-workbench-on-aws/blob/b99092458c7ee11f9b7540f7d1bbf898dd90744f/main/integration-tests/__test__/api-tests/common/workspace-types/configurations/get-configuration.test.js#L135-L141)

To nest ETC inside of ET, you can follow the pattern [here](https://github.com/awslabs/service-workbench-on-aws/blob/9e224ff1aee34ccbdea14f330c7c840a9a39b125/main/integration-tests/support/resources/workspace-types/workspace-type.js#L36)

This way you don't have to add the `parentId` attribute to `collectionResource`

Reply:
got it , I will modify it to use that form